### ISSUE TITLE
REQ: Suggestions? Discussions? Feature list?

### ISSUE THREAD

**Issue by xxxserxxx** [simplex-chat/simplex-chat#836]
There's no group chat published for simplex-chat, so I'm posting here.

Would you be able to start a github discussion (since you're already using github) around feature sets? In particular, I'd like to know whether some things are on the roadmap, are considered out-of-scope (anti-features), or available (but maybe harder to find). I think a github discussion, or even just a wiki page, would be useful for users.

I'm personally interested in the following topics:

1. What's implemented now is one-time invitations. Will there ever be sticky invitations? The main use case for this would be "Contact Me" links, e.g. embedded in email signatures or web pages where anonymity is not crucial.
2. Group chats. Currently not possible with one-time invites, and maybe considered out-of-scope for simplex.
3. Developer thinking about battery issues. There has been work around battery optimization for the Android app, and there's at least one ticket about it, but it'd be nice to know if there are concrete plans and the developer thinking around this topic. Currently, I'm unable to use notifications with Simplex because of the battery use, and it'd be comforting to know that y'all have some ideas on how this can be improved.

To summarize: I'm requesting a wiki page with a list of features, each tagged with Implemented, Anti-feature, or On roadmap. It'd be a bonus to address some of the FAQ questions, like any plans to improve battery life on mobile.

**Comment by epoberezkin** [comment]
@xxxserxxx thank you

> Would you be able to start a github discussion

I thinks issues could be used as discussions, I've seen projects using both, and it fragments the conversation - so issues are ok for that... Maybe I am wrong :)

> Will there ever be sticky invitations?

Assuming you want links that can be used multiple times, they already exist. In terminal app it's /ad command, in mobile - it's "user address" in the settings. We will be changing the UX there to make it easier to understand, as quite a few users are confused about the difference between "1-time links" (we are considering to call them "burner addresses") and "user addresses" (which we then will call "permanent addresses").

> The main use case for this would be "Contact Me" links, e.g. embedded in email signatures or web pages...

Exactly - I have it on my linkedin profile and email signature, you can create one too. I should actually add it to GitHub profile as well.

> ... where anonymity is not crucial.

I am actually not convinced that it has lower privacy, it's a trade-off. On one hand, people are connecting via the same link and it can potentially be correlated, depending on how it is shared. On another hand, the address of the actual queue used for messaging is passed in encrypted form (via the "contact queue").

> Group chats. Currently not possible with one-time invites, and maybe considered out-of-scope for simplex.

https://www.reddit.com/r/SimpleXChat/comments/w1cg6x/chat_groups_accepting_group_invitations_via_ui_is/
https://github.com/simplex-chat/simplex-chat/blob/ep/blog-v31/blog/20220723-simplex-chat-v3.1-tor-groups-efficiency.md

> battery issues.

See same post above. We will be optimising it further. We are also considering adding push notificaitions to android app (same is on iOS) that will be offered as an alternative to an always-on background service. Battery usage varies very widely between users, many users reported under 3% usage per day, and several reported 15%+ usage per day. My understanding is that it is hugely affected by the stability and the latency of network connections, we have several ideas how to reduce battery consumption on bad networks

>  I'm requesting a wiki page with a list of features, each tagged with Implemented, Anti-feature, or On roadmap. It'd be a bonus to address some of the FAQ questions, like any plans to improve battery life on mobile.

Most of it is covered in README page (with the exception of "Anti-feature" - we don't have any strong "no"s right now, and we don't have a firm stance against any ideas), and we might be splitting it to several pages. I am not a fan of using GitHub wiki, as it is outside of source control, so it's more difficult to discuss and peer-review.

**Comment by xxxserxxx** [comment]
Thank you; that is all very clear. I appreciate the response.

**Comment by xxxserxxx** [comment]
> I thinks issues could be used as discussions, I've seen projects using both, and it fragments the conversation - so issues are ok for that... Maybe I am wrong :)

Tickets are fine; I see a lot of devs preferring to keep ticket queues clean -- however you want to do it is fine!

> > Will there ever be sticky invitations?
>
> Assuming you want links that can be used multiple times, they already exist. 

That's perfect!  The documentation only mentions the one-time invites, and I didn't see `/ad` elsewhere, but that's exactly what I was looking for, thank you!

> > ... where anonymity is not crucial.
> 
> I am actually not convinced that it has lower privacy, it's a trade-off. 

Sure. If you post it in an Onion page, or on Freenet, or a friend posts it on your behalf, then anonymity is protected. For my use case, I was thinking more about .sigs and "Contact me" web pages, where who I am is already exposed.

> > Group chats. Currently not possible with one-time invites, and maybe considered out-of-scope for simplex.
> https://www.reddit.com/r/SimpleXChat/comments/w1cg6x/chat_groups_accepting_group_invitations_via_ui_is/ 
> https://github.com/simplex-chat/simplex-chat/blob/ep/blog-v31/blog/20220723-simplex-chat-v3.1-tor-groups-efficiency.md

I hadn't seen the blog, thanks! And you have a reddit sub -- that's great!

> > battery issues.
> 
> See same post above. We will be optimising it further.

I found one or two commits addressing battery, and I appreciate that.

>My understanding is that it is hugely affected by the stability and the latency of network connections, we have several ideas how to reduce battery consumption on bad networks

This is a common problem for chat clients. DHT-based systems (like Tox), in particular, struggle with keeping the battery use down.

One last question: I've been browsing around in https://github.com/simplex-chat/simplex-chat/tree/stable/docs, and in particular the `rfcs` directory. Do you plan on writing a spec for the protocol at some point, so people can write compatible libraries in other languages? The Haskell source is a good start, but lacks contextual description.  The `mmd` sequence diagrams are great, and would be helpful if they covered more of the protocol.

**Comment by epoberezkin** [comment]
> Do you plan on writing a spec for the protocol at some point, so people can write compatible libraries in other languages? 

There are actually 3 levels of the protocol (or even 4, depending what you count as protocol):

1. Low level SMP (SimpleX Messaging Protocol) – it defines how low level SimpleX clients communicate with the servers and each other. It's well documented here: https://github.com/simplex-chat/simplexmq/blob/stable/protocol/simplex-messaging.md Anybody who wants to implement the server (which is quite viable, as the protocol is quite small), e.g. in C or in Rust, would use it – we will support this effort.
2. SMP Agent protocol - middle level protocol that defines how clients manage encryption, introductions and duplex connections. It's documented here: https://github.com/simplex-chat/simplexmq/blob/stable/protocol/agent-protocol.md
3. SimpleX Chat Protocol - just published the first draft [here](https://github.com/simplex-chat/simplex-chat/blob/master/docs/protocol/simplex-chat.md).
4. Not exactly a protocol, more like an API that could be used by whoever uses SimpleX library to embed chat functionality into another app – not documented yet, but will be soon.
